src/vit-servicing-station/vit-servicing-station-tests/src/tests/raw_snapshot.rs

```diff
@@ -0,0 +1,361 @@
+
+    rest_client.put_raw_snapshot(&first_raw_snapshot).unwrap();
+
+    let second_raw_snapshot = RawSnapshotUpdater::from(first_raw_snapshot.clone())
+        .with_tag("fund9")
+        .build();
+
+    rest_client.put_raw_snapshot(&second_raw_snapshot).unwrap();
+
+    let assigner = RepsVotersAssigner::new(
+        first_raw_snapshot.content.direct_voters_group.unwrap(),
+        first_raw_snapshot.content.representatives_group.unwrap(),
+    );
+
+    let first_snapshot = Snapshot::from_raw_snapshot(
+        first_raw_snapshot.content.snapshot,
+        first_raw_snapshot.content.min_stake_threshold,
+        first_raw_snapshot.content.voting_power_cap,
+        &assigner,
+    )
+    .unwrap()
+    .to_full_snapshot_info();
+
+    for (idx, entry) in first_snapshot.iter().enumerate() {
+        let voting_power = VotingPower::from(entry.clone());
+        let voter_info = rest_client
+            .voter_info(&first_raw_snapshot.tag, &entry.hir.voting_key.to_hex())
+            .unwrap();
+        assert_eq!(
+            vec![voting_power.clone()],
+            voter_info.voter_info,
+            "wrong data for entry idx: {}",
+            idx
+        );
+        assert_eq!(
+            first_raw_snapshot.content.update_timestamp, voter_info.last_updated,
+            "wrong timestamp for entry idx: {}",
+            idx
+        );
+
+        let voter_info = rest_client
+            .voter_info(&second_raw_snapshot.tag, &entry.hir.voting_key.to_hex())
+            .unwrap();
+        assert_eq!(
+            vec![voting_power],
+            voter_info.voter_info,
+            "wrong data for entry idx: {}",
+            idx
+        );
+        assert_eq!(
+            second_raw_snapshot.content.update_timestamp, voter_info.last_updated,
+            "wrong timestamp for entry idx: {}",
+            idx
+        );
+    }
+}
+
+#[test]
+pub fn import_malformed_raw_snapshot() {
+    let temp_dir = TempDir::new().unwrap();
+    let (server, data) = quick_start(&temp_dir).unwrap();
+    let rest_client: RawRestClient = server.rest_client_with_token(&data.token_hash()).into();
+
+    let raw_snapshot = RawSnapshot::default();
+    let mut content = serde_json::to_string(&raw_snapshot.content).unwrap();
+    content.pop();
+    assert!(rest_client
+        .put_raw_snapshot(&raw_snapshot.tag, content)
+        .unwrap()
+        .status()
+        .is_client_error());
+}
+
+#[test]
+pub fn import_big_raw_snapshot() {
+    let temp_dir = TempDir::new().unwrap();
+    let (server, data) = quick_start(&temp_dir).unwrap();
+    let mut rest_client = server.rest_client_with_token(&data.token_hash());
+    rest_client.set_timeout(Duration::new(600, 0));
+
+    let raw_snapshot = RawSnapshotBuilder::default()
+        .with_tag("big")
+        .with_voting_registrations_count(100_000)
+        .build();
+
+    rest_client.put_raw_snapshot(&raw_snapshot).unwrap();
+
+    let assigner = RepsVotersAssigner::new(
```

Comment:
another NITPICK. i know that RepsVoterAssigner is dev struct so we should not really touched it. But we can make an extension method for it or implement Into trait and nicely convert raw_snapshot to assigner. I would see something like
```
impl TryFrom<RawSnapshot> for RepsVoterAssigner {
  type Error = ...;

  fn from(snapshot: RawSnapshot) -> Return<Self,Self::Error> {
        Self {
           raw_snapshot.content.direct_voters_group.ok_or(...)?,
           raw_snapshot.content.representatives_group..ok_or(...)?,
        }
  }
}
```

then we can use it like:

`let assigner = raw_snapshot.clone().try_into().unwrap()`

Reply:
Done